pandas/io/formats/printing.py

```diff
@@ -204,11 +204,11 @@ def as_escaped_string(
         thing: Any, escape_chars: EscapeChars | None = escape_chars
     ) -> str:
         translate = {"\t": r"\t", "\n": r"\n", "\r": r"\r"}
-        if isinstance(escape_chars, dict):
+        if isinstance(escape_chars, Mapping):
             if default_escapes:
                 translate.update(escape_chars)
             else:
-                translate = escape_chars
+                translate = escape_chars  # type: ignore[assignment]
```

Comment:
Actually this highlights that if `default_escapes = True`, the `update` call above _might_ fail. In that case maybe casting to a dict is necessary?

Reply:
Sorry if I'm being slow here, can you explain in more detail the exact conditions that would cause failure?

Reply:
This is ignoring `error: Incompatible types in assignment (expression has type "Mapping[str, str] | Mapping[Any, Any]", variable has type "dict[str, str]")  [assignment]`. I thought this was just mypy not understanding whats going on. If I have it right `escape_chars` should only be of type `Mapping[str, str]` here, I'm not sure why it thinks it could be `Mapping[Any, Any]`

Reply:
Ah sorry, my misunderstanding, `translate.update` applies to `translate: dict` and not the `Mapping` change you made.